employee-rostering-gwtui/src/test/java/org/optaweb/employeerostering/gwtui/client/gwtjackson/TestJsonReaderImpl.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaweb.employeerostering.gwtui.client.gwtjackson;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.github.nmorel.gwtjackson.client.stream.JsonReader;
+import com.github.nmorel.gwtjackson.client.stream.JsonToken;
+import com.google.gwt.core.client.JavaScriptObject;
+
+/**
+ * GWT Jackson cannot compile in tests, so we delegate to Jackson.
+ * <p>
+ * Note: gwt-jackson uses nextToken, which Jackson does not have,
+ * thus we use the convention that the current token in the json parser
+ * is the "next token" (i.e. if the current token in the parser is 10,
+ * then nextInt() should return 10)
+ */
+public class TestJsonReaderImpl implements JsonReader {
```

Comment:
It's a pity there is no common interface (I haven't found any) between gwt-jackson and jackson-core, that would suit our needs. That way you could just switch the implementations.